### PR TITLE
Update filtering.md

### PR DESCRIPTION
**Pull request by hunter2009**
1. Add `matching`  to sort matched filters.
2. Adjust dipalay order from `matching` to `excluding`, then can mixed them.

### CODE REVIEW

File: docs/filtering.md [hunk 1]
```diff
@@ -21,8 +21,9 @@ pnpm <command> --filter <package_selector>
 [An article that compares Lerna's filtering to pnpm's](https://medium.com/pnpm/pnpm-vs-lerna-filtering-in-a-multi-package-repository-1f68bc644d6a)
 
 :::
+## matching
```
Comment (juanpicado) [inline docs/filtering.md:24]: Better capitalize here please 🙏🏼

Comment (zkochan) [reply docs/filtering.md:24]: and an empty line is probably needed above

Comment (hunter2009) [reply docs/filtering.md]: OK, updated by issues.

**Review by juanpicado — APPROVED** [review]
LGTM

**Pull request merged**